### ISSUE TITLE
OrmLite populating properties missing in schema with non default values

### ISSUE THREAD

**Issue by mayerwin** [ServiceStack/Issues#384]
Please see below code to reproduce what seems to be a buggy behavior:

        [Alias("MyDTO")]
        public class TypeV1 {
            public string TextProperty { get; set; }
        }

        [Alias("MyDTO")]
        public class TypeV2 {
            public string TextProperty { get; set; }
            public string TextProperty2 { get; set; }
            public HashSet<string> HashSetProperty { get; set; }
        }

        public static void OrmLiteBug() {
            var fp = "test.sqlite";
            var v1 = new TypeV1();
            IO.DeleteIfExists(fp);
            using (var db = SqliteFactory.Open(fp)) {
                db.Save(v1);
                var v2 = db.Get<TypeV2>().Single();
                Console.WriteLine(v2.Dump());
            }
        }

When a property is added to a DTO, but a table does not contain a corresponding column, the expected behavior when querying an item from the table would be for OrmLite to simply populate the new property with its default value. In the case of string and Hashset<string>: null.

Instead, the above code will output the property name (!) as the default value for the string, and a HashSet containing as single item the property name (!) for HashSet<string>(): 

```{
	TextProperty2: TextProperty2,
	HashSetProperty: 
	[
		HashSetProperty
	]
}```

**Comment by mythz** [comment]
To start with each Model needs a Primary Key which if you don't specify one with `[PrimaryKey]`, `[AutoIncrement]` attributes or `Id` property name it defaults to the First property in the type. Inserting a `null` primary key should fail in most RDBMS's, although Sqlite isn't strict and allows it.

Next, you can't read into a Type with **more** properties (only a type with a subset of the properties) since OrmLite generates the SQL that lists every property into the type it's selecting into, so the query this generates is this:

    SELECT "TextProperty", "TextProperty2", "HashSetProperty" FROM "MyDTO"

Again this should fail in most RDBMS's since these columns don't exist. SQLite is buggy here and should throw but instead it returns columns that don't match as literal strings which are mapped to the matching properties.

In summary this is illegal usage that's would fail in a proper RDBMS but Sqlite has buggy implementation that results in the behavior you're seeing.
